File: ios-native/IronCore/Models/UserProfile.swift
import Foundation
import FirebaseFirestore

/// Extended profile at `users/{userId}/data/profile`
/// Single document per user with all detailed profile data
struct UserProfile: Codable, Identifiable {
    var id: String { userId }
    var userId: String
    var photoURL: String?
    var schemaVersion: Int

    // Streaks
    var currentStreak: Int
    var streakMultiplier: Double
    var lastStreakUpdate: String?    // ISO8601
    var streakShields: Int
    var shieldActiveUntil: String?   // ISO8601

    // XP & Progression
    var xp: Int
    var xpHistory: [String: XPHistoryEntry]?

    // Daily Rewards
    var dailyRewardsClaimed: [String: Bool]?  // { "2025-02-25": true }
    var rewardDayIndex: Int
    var lastRewardMonth: String?              // "YYYY-MM"
    var doubleXPTokens: Int

    // Inventory
    var inventory: [InventoryItem]

    // Subscription (duplicate of root doc for consistency)
    var subscription: SubscriptionInfo?
    var isPremium: Bool

    // Nutrition targets
    var dailyCalories: Int?
    var dailyProtein: Int?
    var height: Double?
    var weight: Double?
    var age: Int?
    var gender: String?

    // Onboarding
    var onboarded: Bool?

    // Daily drops
    var dailyDrops: [String: Bool]?  // { "2025-02-25": true }

    static let empty = UserProfile(
        userId: "",
        photoURL: nil,
        schemaVersion: 3,
        currentStreak: 0,
        streakMultiplier: 1.0,
        lastStreakUpdate: nil,
        streakShields: 0,
        shieldActiveUntil: nil,
        xp: 0,
        xpHistory: nil,
        dailyRewardsClaimed: nil,
        rewardDayIndex: 0,
        lastRewardMonth: nil,
        doubleXPTokens: 0,
        inventory: [],
        subscription: nil,
        isPremium: false,
        dailyCalories: nil,
        dailyProtein: nil,
        height: nil,
        weight: nil,
        age: nil,
        gender: nil,
        onboarded: nil,
        dailyDrops: nil
    )
}

struct XPHistoryEntry: Codable {
    var amount: Int
    var reason: String
}

struct InventoryItem: Codable {
    var item: String
    var boughtAt: String?       // ISO8601 timestamp
    var receivedAt: String?     // ISO8601 timestamp (for rewards)
}

File: android-native/app/src/main/java/com/ironcore/fit/data/model/UserProfile.kt
package com.ironcore.fit.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.PropertyName
import com.google.firebase.firestore.ServerTimestamp

/**
 * Firestore path: /users/{userId}/data/profile
 * Mirrors React useFitnessData.js profile schema exactly.
 */
data class UserProfile(
    val schemaVersion: Int = 1,
    val userId: String = "",
    val photoURL: String = "",
    val xp: Long = 0,
    val level: Int = 0,
    val currentStreak: Int = 0,
    val longestStreak: Int = 0,
    val streakFreezeCount: Int = 1,
    val streakShields: Int = 0,
    val shieldActiveUntil: String? = null,
    val doubleXPTokens: Int = 0,
    val lastLoginAt: Timestamp? = null,
    val lastStreakUpdateAt: Timestamp? = null,
    val dailyDrops: Map<String, Boolean> = emptyMap(),
    val dailyRewardsClaimed: Map<String, Boolean> = emptyMap(),
    val rewardDayIndex: Int = 0,
    val lastRewardMonth: String = "",
    val subscription: SubscriptionInfo? = null,
    val isPremium: Boolean = false,
    val guildId: String? = null,
    val guildRole: String? = null,
    val stackConfig: List<String> = emptyList(),
    val inventory: List<InventoryItem> = emptyList()
)

data class SubscriptionInfo(
    val planId: String = "free",
    val status: String = "expired",
    val startDate: String = "",
    val expiryDate: String = "",
    val paymentId: String = "",
    val orderId: String = "",
    val updatedAt: Timestamp? = null
)

data class InventoryItem(
    val item: String = "",
    val boughtAt: Timestamp? = null,
    val receivedAt: Timestamp? = null
)
